### PR TITLE
Fixed RecursionError caused by giving `config_from_object` nested mod…

### PR DESCRIPTION
**Pull request by frolenkov-nikita**
Fixed RecursionError caused by giving `config_from_object` nested module that does not exist (#8517)

## Description

The problem is described by new unit test: t/unit/app/test_app.py::test_config_form_object__module_attr_does_not_exist

When you provide a string with valid module name but invalid submodule (attr) to `app.config_from_object` the 3rd party util `kombu.utils.imports.symbol_by_name` would raise `AttributeError` which would be "handled" by `PendingConfiguration(UserDict)` and would cause infinite recursion. This happens during the "lazy initialization" of the `app.conf` when `app.conf` attr is accessed.

The fix is to catch `AttributeError` and raise `ModuleNotFoundError` which would be propagated correctly and would provide the real error message to the end user.

I don't feel very good re: this fix, I am new to this project, so your criticism is highly desired.

### CODE REVIEW

File: t/unit/app/test_app.py
Comment (Nusnus) [inline t/unit/app/test_app.py:703]: Shouldn't this and the next assertion be outside the `with` scope?

**Review by Nusnus — COMMENTED** [review]
Nice work!
Check out my comment

File: celery/app/base.py
Comment (Nusnus) [inline celery/app/base.py:984]: You can use `raise from` e.g raise Module..() from e

File: celery/app/base.py
Comment (Nusnus) [inline celery/app/base.py:980]: Not critical, but more common `err` than just `e`

**Comment by frolenkov-nikita** [comment]
Thank you very much, I've incorporated your feedback.

File: t/unit/app/test_app.py
Comment (auvipy) [reply t/unit/app/test_app.py:706]: is this resolved? @frolenkov-nikita

**Review by auvipy — COMMENTED** [review]
seems better now. waiting for another round of review from Nusnus

**Review by auvipy — APPROVED** [review]

**Review by Nusnus — APPROVED** [review]
Sorry for the delay, good work!

**Pull request merged**
